### PR TITLE
Update newsletter form to use Hubspot

### PR DESCRIPTION
**Pull request by ceolson01**
## Description

This PR updates the newsletter form to use Hubspot. It also introduces a new hook which reduces some boilerplate when working with Hubspot forms.

**New stuff** ✨

* `useHubspotForm` hook

**Changes** 🏗

* Update all hubspot forms to use the new `useHubspotForm` hook
  * `NewsletterCta` (mailchimp -> Hubspot)
  * `WebsiteLayout.Footer` (mailchimp -> Hubspot)
  * `GetEarlyAccess.Form`

### CODE REVIEW

File: src/hooks/useHubspotForm.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  portalId: string,
+|};
+
+type FormData = {
+  /*
+   * Hubspot's field name (same as the `input` `name` attribute on Hubspot Forms)
+   *
+   * e.g. { 'company_name': 'ABC', 'company_size': '6' }
+   */
+  [fieldName: string]: string,
```
Comment (ryanchristo) [inline src/hooks/useHubspotForm.js]: :ok_hand: Looks like it will always be a string.

File: src/hooks/useHubspotForm.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+};
+
+type HubspotFieldsData = Array<{
```
Comment (ryanchristo) [inline src/hooks/useHubspotForm.js]: Should this be exact typing?

File: src/hooks/useHubspotForm.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  name: string,
+  value: string,
+}>;
+
+type Status = {| error: string |} | {| response: Object |} | void;
```
Comment (ryanchristo) [inline src/hooks/useHubspotForm.js:26]: :ok_hand: Probably not necessary to define the response object.

File: src/hooks/useHubspotForm.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+type OnError = () => void;
+type OnSuccess = () => void;
+
+type HookReturn = {
```
Comment (ryanchristo) [inline src/hooks/useHubspotForm.js]: Should this be exact typing?

File: src/hooks/useHubspotForm.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+const useHubspotForm = ({
+  formGuid,
+  pageName,
+  pageUri,
+  portalId,
+}: FormConfig): HookReturn => {
+  // eslint-disable-next-line max-len
+  const endpoint = `https://api.hsforms.com/submissions/v3/integration/submit/${portalId}/${formGuid}`;
+
+  const [status, setStatus] = useState<Status>();
```
Comment (ryanchristo) [inline src/hooks/useHubspotForm.js:50]: I like how you combined `error` and `response` into `Status` here.

File: src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx [hunk 6]
```diff
@@ -39,29 +40,27 @@ const Footer = () => {
     [],
   );
 
+  const { error, response, submitForm } = useHubspotForm({
```
Comment (ryanchristo) [inline src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx:43]: :ok_hand: Not necessary to extract values but `formGuid` (x2) and `portalId` (x3) are being reused.

File: src/modules/pages/components/Dashboard/Admin/AddPayment/AddPayment.jsx [hunk 7]
```diff
@@ -110,7 +110,9 @@ const AddPayment = ({ colonyClient, network, user, wallet }: Props) => {
       // eslint-disable-next-line no-undef
       fetch(
         // eslint-disable-next-line max-len
-        `${server}/api/contribution?sessionID=${user.session.id}&address=${wallet.address}&network=${network.slug}`,
+        `${server}/api/contribution?sessionID=${user.session.id}&address=${
```
Comment (ryanchristo) [inline src/modules/pages/components/Dashboard/Admin/AddPayment/AddPayment.jsx]: If we change to this, we no longer need `eslint-disable-next-line`. My editor still prefers that former. :thinking:

File: src/modules/pages/components/Dashboard/Admin/AddTask/AddTask.jsx [hunk 8]
```diff
@@ -107,7 +107,9 @@ const AddTask = ({ colonyClient, network, user, wallet }: Props) => {
       // eslint-disable-next-line no-undef
       fetch(
         // eslint-disable-next-line max-len
-        `${server}/api/contribution?sessionID=${user.session.id}&address=${wallet.address}&network=${network.slug}`,
+        `${server}/api/contribution?sessionID=${user.session.id}&address=${
```
Comment (ryanchristo) [inline src/modules/pages/components/Dashboard/Admin/AddTask/AddTask.jsx]: If we change to this, we no longer need `eslint-disable-next-line`. My editor still prefers that former. :thinking:

**Review by ryanchristo — APPROVED** [review]
Nice work! Some questions and comments but tested and preapproved. :+1:

File: src/hooks/useHubspotForm.js (hunk 2 above)
Comment (rdig) [reply src/hooks/useHubspotForm.js]: I would argue yes. At least for the input data.

On the return messages / object, I'm with you, just some generic types should suffice.

File: src/hooks/useHubspotForm.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      })),
+    [],
+  );
+
+  const submitForm = useCallback(
+    (formData: FormData, onSuccess?: OnSuccess, onError?: OnError) => {
+      const fetchOptions = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
```
Comment (rdig) [inline src/hooks/useHubspotForm.js]: I'm not sure what Hubspot expects, but if we bothered to set the content type, shouldn't we also set the content encoding ? _(ie: `utf-8`)_

File: src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx [hunk 10]
```diff
@@ -39,29 +40,27 @@ const Footer = () => {
...
+    formGuid: 'd0e98a7b-aba5-412b-8c12-ace04835bad9',
+    pageName: '*Any Page (Footer)',
+    pageUri: 'https://colony.io/*',
+    portalId: '4846129',
```
Comment (rdig) [inline src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx:47]: Shouldn't these values be passed in as an `env` variable at build time rather then hardcoding them in ?

Either via a `.env` file or just process variables: `PORTAL_ID=4846129 yarn build`

File: src/modules/pages/components/Website/EarlyAccess/Form.jsx [hunk 11]
```diff
@@ -88,13 +89,12 @@ const validationSchema = yup.object().shape({
...
-
-  const portalId = '4846129';
-  const formGuid = '8e34f607-d535-42eb-8b34-38fb6b31ab49';
-  // eslint-disable-next-line max-len
-  const endpoint = `https://api.hsforms.com/submissions/v3/integration/submit/${portalId}/${formGuid}`;
+  const { error, response, submitForm } = useHubspotForm({
+    formGuid: '8e34f607-d535-42eb-8b34-38fb6b31ab49',
+    pageName: 'Get Early Access',
+    pageUri: `https://colony.io${PAGE_GET_EARLY_ACCESS}`,
+    portalId: '4846129',
```
Comment (rdig) [inline src/modules/pages/components/Website/EarlyAccess/Form.jsx:96]: See above about using `env` variables

**Review by rdig — CHANGES_REQUESTED** [review]
I'm good with most of this, except the hardcoded id values.

Other then that, very nice work on that custom hook! :tada:

File: src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx (hunk 10 above)
Comment (ceolson01) [reply src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx:47]: They certainly *could* be. But this isn't exactly a privacy concern, as both `portalId` and `formGuid` are publicly available at HubSpot (all forms are public). What would you prefer?

File: src/modules/pages/components/Dashboard/Admin/AddPayment/AddPayment.jsx (hunk 7 above)
Comment (ceolson01) [reply src/modules/pages/components/Dashboard/Admin/AddPayment/AddPayment.jsx]: Hm, not sure how this ended up in here - I've updated this to match what `yarn lint` yields

File: src/hooks/useHubspotForm.js (hunk 2 above)
Comment (ceolson01) [reply src/hooks/useHubspotForm.js]: Nice nice! All this typescript in my life is making me forget to seal it up on old flow projects 😉

File: src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx (hunk 10 above)
Comment (rdig) [reply src/modules/layouts/components/WebsiteLayout/Footer/Footer.jsx:47]: If they're public then they're fine as-is. :+1:

**Review by rdig — APPROVED** [review]
Good to go from my end! :tada:

**Pull request merged**